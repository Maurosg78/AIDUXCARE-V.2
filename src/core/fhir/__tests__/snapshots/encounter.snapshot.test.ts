// src/core/fhir/__tests__/snapshots/encounter.snapshot.test.ts
import { describe, it, expect } from 'vitest';
import { internalEncounterMock } from '../__fixtures__/internal.mock';
// Importar desde el index principal
import { toFhirEncounter } from '../../index';

describe('FHIR Encounter snapshot', () => {
  it('matches expected snapshot for an ambulatory encounter', () => {
    const fhirEncounter = toFhirEncounter(internalEncounterMock, { profile: 'US_CORE' });
    // Comprobaciones mínimas coherentes con el Sprint 2
    expect(fhirEncounter.resourceType).toBe('Encounter');

    // Asegura que class.code sea el mapeo correcto (AMB)
    // según la regla que definimos (ambulatory → AMB)
    expect(fhirEncounter.class?.code).toBe('AMB');

    // Referencias coherentes (Patient/Practitioner/Organization si tu adapter las genera)
    // Nota: Ajusta según tu implementación (reference vs. display)
    if (Array.isArray(fhirEncounter.participant)) {
      expect(fhirEncounter.participant[0]).toBeDefined();
    }

    expect(fhirEncounter).toMatchSnapshot();
  });
});
