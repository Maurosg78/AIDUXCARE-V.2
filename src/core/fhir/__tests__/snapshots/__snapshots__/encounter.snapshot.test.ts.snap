// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FHIR Encounter snapshot > matches expected snapshot for an ambulatory encounter 1`] = `
{
  "class": {
    "code": "AMB",
    "display": "Ambulatory",
    "system": "http://terminology.hl7.org/CodeSystem/v3-ActCode",
  },
  "id": "enc-1001",
  "meta": {
    "lastUpdated": "2025-08-11T20:42:02.255Z",
    "profile": [
      "http://hl7.org/fhir/us/core/StructureDefinition/us-core-encounter",
    ],
  },
  "participant": [
    {
      "individual": {
        "display": "Provider pr-001",
        "reference": "urn:uuid:0ba807a5-0ba8-07a5-0ba8-0ba807a50ba8",
      },
      "type": [
        {
          "coding": [
            {
              "code": "ATND",
              "display": "attender",
              "system": "http://terminology.hl7.org/CodeSystem/v3-ParticipationType",
            },
          ],
        },
      ],
    },
  ],
  "period": {
    "end": "2025-08-01T10:10:00Z",
    "start": "2025-08-01T09:30:00Z",
  },
  "reasonCode": [
    {
      "coding": [
        {
          "code": "unknown",
          "display": "Dolor de hombro derecho tras esfuerzo",
          "system": "http://snomed.info/sct",
        },
      ],
      "text": "Dolor de hombro derecho tras esfuerzo",
    },
  ],
  "resourceType": "Encounter",
  "status": "finished",
  "subject": {
    "display": "Patient pt-0001",
    "reference": "urn:uuid:36ca3300-36ca-3300-36ca-36ca330036ca",
  },
  "type": [
    {
      "coding": [
        {
          "code": "AMB",
          "display": "Ambulatory",
          "system": "http://terminology.hl7.org/CodeSystem/v3-ActCode",
        },
      ],
      "text": "Dolor de hombro derecho tras esfuerzo",
    },
  ],
}
`;
